### PR TITLE
auto scaler will work only with jenkins node registred with label

### PR DESCRIPTION
**Pull request by Shareed2k**

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -74,8 +74,8 @@ var (
 			Required:   true,
 		},
 		{
-			Name:     "exclude_nodes_by_label",
-			Help:     "exclude jenkins nodes by label",
+			Name:     "find_by_node_label_name",
```
Comment (shamil) [inline pkg/config/config.go:77]: Suggest this name for the param. By the way the right terminalogy for `jenkins` is `agent` and not `nodes`.

```suggestion
			Name:     "agent_with_label",
```

File: pkg/scaler/scaler.go [hunk 2]
```diff
@@ -159,15 +161,14 @@ func (s *Scaler) Do(ctx context.Context) {
 
 	logger := s.logger
 
-	usage, err := s.client.GetCurrentUsage(ctx)
+	// get backend instances to calculate usage
+	ins, err := s.backend.Instances()
```
Comment (shamil) [inline pkg/scaler/scaler.go:165]: should we store the instances in the `Scaler` struct? So they can be used later, in `sclaeDown` method for example or `removeNode`?

**Review by shamil — CHANGES_REQUESTED** [review]

File: pkg/scaler/scaler.go (hunk 2 above)
Comment (Shareed2k) [reply pkg/scaler/scaler.go:165]: instances is already passed to `removeNode` func no need to store it

File: pkg/config/config.go (hunk 1 above)
Comment (Shareed2k) [reply pkg/config/config.go]: Screenshot 2023-01-12 at 9 50 07
 i"m not sure, there is know meaning of agents in jenkins only nodes,

File: pkg/scaler/scaler.go (hunk 2 above)
Comment (shamil) [reply pkg/scaler/scaler.go:165]: That what I mean, when you have instance method you can store the instances in the instance properties and avoid passing to the methoeds. This is just preferred (not enforced) in OOP.

File: pkg/config/config.go [hunk 3]
```diff
@@ -74,8 +74,8 @@ var (
 			Required:   true,
 		},
 		{
-			Name:     "exclude_nodes_by_label",
-			Help:     "exclude jenkins nodes by label",
+			Name:     "node_with_label",
```
Comment (shamil) [inline pkg/config/config.go:77]: Should be in plural, I had a typo in my suggestion

```suggestion
			Name:     "nodes_with_label",
```

**Review by shamil — CHANGES_REQUESTED** [review]

File: pkg/scaler/scaler.go (hunk 2 above)
Comment (Shareed2k) [reply pkg/scaler/scaler.go:165]: i"m calling in two places for `backend.Instances()` in `DO` and `GC` func they are in separate goroutines, so what i should do store instances obj only for `DO` and the `GC` will you use as now?

Comment (shamil) [reply pkg/scaler/scaler.go:165]: Lets keep it as is, with go routins you will have to deal concurency.

**Review by shamil — APPROVED** [review]

**Review by HagaiBarel — APPROVED** [review]

**Pull request merged**